File: DumpyServer/Models/DatabaseConnection.cs
using System;

namespace DumpyServer.Models
{
    public class DatabaseConnection
    {
        public int DbInfoId { get; set; }
        public int DuUserId { get; set; }
        public string ConnectionName { get; set; } = string.Empty;
        public string Type { get; set; } = string.Empty;
        public string Host { get; set; } = string.Empty;
        public int Port { get; set; }
        public string Database { get; set; } = string.Empty;
        public string Username { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;
        public bool Ssl { get; set; }
        public string OptionsJson { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}

File: src/models/DatabaseConnection.ts
export interface DatabaseConnection {
  connectionName: string;
  databaseType: string;
  hostName: string;
  portId: number;
  serviceName: string;
  username: string;
  password: string;
  sslRequired: boolean;
  optionsJson?: string;
}

export interface DatabaseConnectionRequest {
  connectionName: string;
  databaseType: string;
  encryptedHostName: string;
  portId: number;
  encryptedServiceName: string;
  encryptedUsername: string;
  encryptedPassword: string;
  sslRequired: boolean;
  optionsJson?: string;
}

export interface DatabaseConnectionResponse {
  dbInfoId: number;
  duUserId: number;
  connectionName: string;
  type: string;
  host: string;
  port: number;
  database: string;
  username: string;
  password: string;
  ssl: boolean;
  optionsJson: string;
  createdAt: string;
  updatedAt: string;
} 